New Blog Post: Reflecting on My Outreachy Journey: Growth, Challenges, and New Beginnings (#6512)
open-telemetry/opentelemetry.io: content/en/blog/2025/outreachy-internship.md

selection for
[Outreachy](https://richardchukwu.hashnode.dev/outreachy-internship-introduction-first-day-of-my-internship#heading-determination).
It was a moment of validation, but also one filled with uncertainty. Could I
truly make an impact in open source? Would I be able to contribute meaningfully
to a project as vast and complex as OpenTelemetry? Now, as I wrap up this
incredible journey, I can say with confidence that the experience has been
transformative in ways I never imagined.

## Overcoming Fear: Imposter Syndrome Didn’t Win

One of my biggest fears stepping into Outreachy was imposter syndrome.
OpenTelemetry is a widely used project, and I worried about keeping up with
seasoned contributors. However, I quickly realized that the open source
community thrives on collaboration, mentorship, and shared learning. My mentor
and the community members were incredibly supportive, always willing to guide me
when I felt stuck. I learned that asking questions and seeking help is not a
weakness — it’s how we grow.

## A Highlight: Seeing My Work Make an Impact

One of the most fulfilling moments of my internship was when my

> [first major contribution](https://github.com/open-telemetry/opentelemetry-erlang/pull/825) — a
> detailed setup guide for OpenTelemetry Erlang contributors was merged. What

Reviewer: ```suggestion
[first major contribution](https://github.com/open-telemetry/opentelemetry-erlang/pull/825)&mdash;a
detailed setup guide for OpenTelemetry Erlang contributors&mdash;was merged. What
```

seemed like a simple task at first turned into a deep dive into improving
developer onboarding, documenting tribal knowledge, and making contributions
more accessible. Seeing community members reference my work and use it to get
started was an incredibly rewarding feeling. It reminded me why I love open

> source — it's about creating something that empowers others.

Reviewer: ```suggestion
source&mdash;it's about creating something that empowers others.
```

> ## Growth in Skills and Confidence

Reviewer: ```suggestion
## Growth in skills and confidence
```

This internship stretched my technical and communication skills in ways I hadn’t
expected. On the technical side, I gained a deeper understanding of:

- OpenTelemetry’s instrumentation and architecture.
- Observability and its role in modern distributed systems.
- Documentation best practices and how to bridge the gap between technical
  complexity and accessibility.

On the communication side, I learned how to:

- Engage in meaningful discussions in GitHub issues and PRs.
- Synthesize feedback from multiple contributors and maintainers.
- Document processes in a way that is both clear and effective for a global
  audience.

Most importantly, Outreachy gave me the confidence to contribute to open source
projects beyond this internship. I now see myself as part of the broader open
source ecosystem, and I’m excited to continue making meaningful contributions.

> ## Project Contributions and Future Directions

Reviewer: I think you could probably merge this section with the previous one, since they overlap a bit.

Author: Alright

During my tenure, I completed the creation of the
[setup guide for contributions on OpenTelemetry](https://github.com/open-telemetry/sig-contributor-experience/issues/31)
and contributed to refining existing documentation. Moving forward, I aim to
continue my involvement with the OpenTelemetry community, focusing on expanding
resources that facilitate contributor engagement.

> ## The Role of My Mentor

Reviewer: ```suggestion
## The role of my mentor
```

My mentor, [Marylia Gutierrez](https://github.com/maryliag), played a crucial
role in my journey. She not only provided guidance on technical issues but also
encouraged me to think critically and independently. She fostered a welcoming
environment where I felt comfortable experimenting, failing, and learning from
my mistakes. Her patience and willingness to share knowledge made a lasting
impact on me.

> ## What’s Next?

Reviewer: ```suggestion
## What’s next?
```

> As I complete my Outreachy internship with CNCF OpenTelemetry, I’m looking ahead

Reviewer: ```suggestion
As I complete my Outreachy internship with the OpenTelemetry project, I’m looking ahead
```

---
title: Reflecting on my Outreachy Journey
linkTitle: Outreachy Internship
date: 2025-03-11
author: '[Richard Chukwu](https://github.com/richardchukwu)'
issue: 6488
sig: Contributor Experience

> cSpell:ignore: # are but Chukwu here Marylia Outreachy that the

Reviewer 2: if you're not ignoring those words, there is no reason to have this here

foundation of my professional aspirations.

I still remember the nervous excitement I felt when I received the
[email](https://x.com/RichardChukwu_/status/1862390957217333461) confirming my
selection for
[Outreachy](https://richardchukwu.hashnode.dev/outreachy-internship-introduction-first-day-of-my-internship#heading-determination).
It was a moment of validation, but also one filled with uncertainty. Could I
truly make an impact in open source? Would I be able to contribute meaningfully
to a project as vast and complex as OpenTelemetry? Now, as I wrap up this
incredible journey, I can say with confidence that the experience has been
transformative in ways I never imagined.

## Overcoming Fear: Imposter Syndrome Didn’t Win

One of my biggest fears stepping into Outreachy was imposter syndrome.
OpenTelemetry is a widely used project, and I worried about keeping up with
seasoned contributors. However, I quickly realized that the open source
community thrives on collaboration, mentorship, and shared learning. My mentor
and the community members were incredibly supportive, always willing to guide me
when I felt stuck. I learned that asking questions and seeking help is not a

> weakness—it’s how we grow.

Reviewer 2: ```suggestion
weakness — it’s how we grow.
```

## A Highlight: Seeing My Work Make an Impact

One of the most fulfilling moments of my internship was when my

> [first major contribution](https://github.com/open-telemetry/opentelemetry-erlang/pull/825)—a

Reviewer 2: ```suggestion
[first major contribution](https://github.com/open-telemetry/opentelemetry-erlang/pull/825) — a
```

detailed setup guide for OpenTelemetry Erlang contributors was merged. What
seemed like a simple task at first turned into a deep dive into improving
developer onboarding, documenting tribal knowledge, and making contributions
more accessible. Seeing community members reference my work and use it to get
started was an incredibly rewarding feeling. It reminded me why I love open

> source—it's about creating something that empowers others.

Reviewer 2: ```suggestion
source — it's about creating something that empowers others.
```

## Growth in Skills and Confidence

This internship stretched my technical and communication skills in ways I hadn’t
expected. On the technical side, I gained a deeper understanding of:

- OpenTelemetry’s instrumentation and architecture.
- Observability and its role in modern distributed systems.
- Documentation best practices and how to bridge the gap between technical
  complexity and accessibility.

On the communication side, I learned how to:

- Engage in meaningful discussions in GitHub issues and PRs.
- Synthesize feedback from multiple contributors and maintainers.
- Document processes in a way that is both clear and effective for a global
  audience.

Most importantly, Outreachy gave me the confidence to contribute to open source
projects beyond this internship. I now see myself as part of the broader open
source ecosystem, and I’m excited to continue making meaningful contributions.

## Project Contributions and Future Directions

During my tenure, I completed the creation of the

> [setup guide for the development of OpenTelemetry](https://github.com/open-telemetry/sig-contributor-experience/issues/31)

Reviewer 2: ```suggestion
[setup guide for contributions on OpenTelemetry](https://github.com/open-telemetry/sig-contributor-experience/issues/31)
```

---
title: Reflecting on my Outreachy Journey
linkTitle: Outreachy Internship

> date: 2025-03-11

Reviewer: ```suggestion
date: 2025-03-24
```

developer onboarding, documenting tribal knowledge, and making contributions
more accessible. Seeing community members reference my work and use it to get
started was an incredibly rewarding feeling. It reminded me why I love open
source&mdash;it's about creating something that empowers others.

This internship stretched my technical and communication skills in ways I hadn’t
expected. On the technical side, I gained a deeper understanding of:

- OpenTelemetry’s instrumentation and architecture.
- Observability and its role in modern distributed systems.
- Documentation best practices and how to bridge the gap between technical
  complexity and accessibility.

On the communication side, I learned how to:

- Engage in meaningful discussions in GitHub issues and PRs.
- Synthesize feedback from multiple contributors and maintainers.
- Document processes in a way that is both clear and effective for a global
  audience.

Most importantly, Outreachy gave me the confidence to contribute to open source
projects beyond this internship. I now see myself as part of the broader open
source ecosystem, and I’m excited to continue making meaningful contributions.

> ## Project Contributions and Future Directions

Reviewer: ```suggestion
## Project contributions and future directions
```
